service/controller/resource/cleanupendpointips/create.go

```diff
@@ -179,7 +179,11 @@ func removeDeadIPFromEndpoints(endpoints *corev1.Endpoints, nodes []corev1.Node,
 		found := false
 		// check if the ip belongs to any k8s node
 		for _, node := range nodes {
-			if node.Labels["ip"] == ip.IP {
+			nodeIP, err := key.NodeInternalIP(node)
```

Comment:
At which point is the IP written to the status? Is there a different order of execution now or has everything essentially stayed the same?

I am concerned that we might introduce a hidden dependency on another controller (which likely already exists) but are not certain of the timings.

Reply:
When we start kubelet, we pass the IP as  `--node-ip` and `--node-labels` so I assumed that kubelet is somehow responsible for populating both. But I'll dig deeper into that.

Reply:
Sorry I wasn't clear:
I am not concerned about the node IP on the k8s object but the status vs label thing

Reply:
Setting the node status is one of the first things that kubelet does once it starts running ([click](https://github.com/kubernetes/kubernetes/blob/1df1f882c4421d313abb468fef5c15f89983471c/pkg/kubelet/kubelet.go#L1394)). It iterates and calls a bunch of setters and [NodeStatus](https://github.com/kubernetes/kubernetes/blob/1df1f882c4421d313abb468fef5c15f89983471c/pkg/kubelet/nodestatus/setters.go#L60) is one of them. I haven't checked who sets the labels but I'll be surprised if it's happening earlier. I also tested this on `gorgoth` and it sounded happy.